plugins-structure/README.md

```diff
@@ -0,0 +1,579 @@
+> but would be episodically timestamped onto the Decred blockchain" - _Politeia:
+> Pi 2021 Q3 (Proposal System)_ [[2]](#2).
+
+However, the stored data could be any set of data, not exclusively proposals,
+which could also be used as a more generic storage, such as contractors
+payments management, proposals discussions, etc.
+
+The current design would allow developers to create new plugins to interact,
+extend and manage the Record functionality. Therefore, we can use Politeia
+[[1]](#1) architecture to create more applications, such as
+[Decred's Proposal system (Pi)](https://proposals.decred.org) and
+[Decred's Contractor Management System (CMS)](https://cms.decred.org/)
+_(CMS not implemented on the new architecture yet)_.
+
+PoliteiaGUI is the Guest User Interface side of Politeia, which stands for the
+same principles of a **plugin-composable architecture**. It is responsible for
+displaying the multiple stored Records on a user-friendly interface, helping
+people who aren't familiar with programming and CLI management also interact
+and use the functionalities provided by the regarded system.
+
+The architecture consists in a composition of one or more plugins that extend
+the Core functionalities. For example, plugins can enhance Records by providing
+extra information, data, statuses, etc. using its own isolated design. This will
+provide a reusable structure for multiple applications, which will provide a
+better code maintainance and scalability.
+
+On the Frontend side, it means we can provide different UXs depending on the
+plugins combinations. We can integrate multiple plugins for customizing some
+app, and still not trigger collateral effects on other plugins.
+
+## GUI Composable Plugin Architecture
+
+The GUI architecture is a mono-repository structure, composed by
+[micro-frontends](https://micro-frontends.org/) [[3]](#3), managed by
+[Lerna](https://lerna.js.org/) and
+[Yarn Workspaces](https://yarnpkg.com/features/workspaces), containing two
+different types of workspaces: _Apps_ and _Packages_.
+
+- Apps: workspace for **application shells**. It's used for micro-frontends
+  integration, in our case, plugins and core, into one user application.
+- Packages: workspace for plugins. Each **plugin** has its own isolated config,
+  and can compose the application shell by extending the **core** package and
+  adding new features, new routes, and also a custom UI.
+
+The following diagram displays an overview of the regarded plugin architecture
+for the proposals (Politeia) application:
+
+<p align="center">
+  <img src="../docs/assets/ps-overview-politeia.svg" />
+</p>
+
+Politeia is the application-shell. Pi, Ticketvote, Comments are plugins that
+extend the core Record package.
+
+> The User layer is currently under development and you can follow its progress
+> on [(decred/politeia#1479) politeiaww: Add user
+> plugins](https://github.com/decred/politeia/issues/1479).
```

Comment:
I don't think we have to mention the under development stage of the user layer because the pgui plugin architecture will be merge only after its completion.